glean-core/android/src/test/java/mozilla/telemetry/glean/GleanTest.kt

```diff
@@ -474,4 +476,27 @@ class GleanTest {
         Glean.setUploadEnabled(true)
         assertTrue(GleanInternalMetrics.os.testHasValue())
     }
+
+    @Test
+    fun `Workers should be cancelled when disabling uploading`() {
+        // Force the MetricsPingScheduler to schedule the MetricsPingWorker
```

Comment:
I think I'd like to see this being more "unit" and less "integration": if we have two `cancel` functions, then we can test each one independently in their modules using the workmanager. Here, we'd just stub out and make sure that `cancel` is called when `setUploadEnabled` is called.

Reply:
If I'm going to make the `cancel` functions static members of their respective classes, it's not really practical to stub out a static method using Mockito so I propose to leave this test as it is and file a follow up bug to investigate the best way to approach this.  I did add unit tests for the cancel functions in each of the test classes associated with the workers.

Reply:
Good point, makes sense.